lib/twingly/url.rb

```diff
@@ -103,11 +108,26 @@ def try_addressable_normalize(addressable_uri)
         raise
       end
 
+      def valid_hostname?(domain)
+        # No need to check the TLD, the public suffix list does that
+        labels = domain.split(DOT)[..-2].map(&:to_s)
+
+        valid = labels.each do |label|
+          break false if label.start_with?(HYPEN)
+          break false if label.end_with?(HYPEN)
+          break false unless label.match?(LETTERS_DIGITS_HYPHEN)
+          true
+        end
```

Comment:
I think you might want `break true` on the last line, or switch to using something like `#all?` instead. Note the subtle difference, in particular for 10 and 12:

```ruby
[10] pry(main)> [1,2,3].each { |num| break true }
=> true
[11] pry(main)> [1,2,3].each { |num| break false }
=> false
[12] pry(main)> [1,2,3].each { |num| true }
=> [1, 2, 3]
```

Reply:
True, but I don't think `break true` is good either, need to check all labels

I wanted to stop checking labels when one was found to be not valid

Reply:
I guess the  final `true` isn't actually needed at all? I don't mind it though, as it could improve readability.

I like @walro's suggestion with `#all?` however!

```suggestion
        valid = labels.all? do |label|
          label.match?(LETTERS_DIGITS_HYPHEN) &&
            !label.start_with?(HYPEN) &&
            !label.end_with?(HYPEN)
        end
```

Reply:
Yes will re-work this